### ISSUE TITLE
Better `immutable` schema check

### ISSUE THREAD

**Issue by thefosk** [Mashape/kong#942]
The `immutable` check has been implemented to avoid changing the value on an update. The current implementation has the side effect of breaking `PUT` requests that don't change the value.

For example if I submit a `PUT` request to change an object, without changing `created_at`, the request will return an error telling me I cannot update `created_at` (and I am not, since I am not changing its value) and the `PUT` request will only work if I remove the `created_at` field from the request (which is against the HTTP spec for `PUT`). If I remove the `created_at` field, the request will go through and work (although in a `PUT` request that would mean **removing** a field).

Basically `immutable` needs to check if the value is still the same, and if it's still the same it should not return any error.
